### ISSUE TITLE
Ensure compatibility with Sync Gateway 2.x

### ISSUE THREAD

**Issue by OldSneerJaw** [Kashoo/synctos#259]
A Sync Gateway 2.0 beta is [available](https://packages.couchbase.com/releases/couchbase-sync-gateway/2.0.0-beta1/couchbase-sync-gateway-enterprise_2.0.0-beta1_x86_64.tar.gz) ([release notes](https://developer.couchbase.com/documentation/mobile/2.0/references/sync-gateway/release-notes/index.html)). Verify that synctos' generated sync functions are compatible with it. Notably SG 2.0 has upgraded to a newer [version](https://github.com/robertkrimen/otto/tree/a813c59b1b4471ff7ecd3b533bac2f7e7d178784) of the Otto JavaScript engine that [claims](https://github.com/robertkrimen/otto/tree/a813c59b1b4471ff7ecd3b533bac2f7e7d178784#caveat-emptor) to be compatible with ECMAScript 5, except for some [noted](https://github.com/robertkrimen/otto/tree/a813c59b1b4471ff7ecd3b533bac2f7e7d178784#regular-expression-incompatibility) RegExp incompatibilities.

Furthermore, since ES5 (and by extension this newer version of otto) does include `JSON.stringify`, the `templates/sync-function/json-stringify-module.js` module should only be used as a fallback for older versions of Sync Gateway. For example, in `templates/sync-function/template.js`:

```
var jsonStringify = (JSON && JSON.stringify) ? JSON.stringify : importSyncFunctionFragment('json-stringify-module.js');
```

Note that the version of Underscore.js (1.4.4) has not changed in the newer version of otto.

**Comment by OldSneerJaw** [comment]
I have performed a number of tests on the Sync Gateway 2.0 beta with sync functions generated by synctos, and it appears to be fully compatible. Pull request #262 includes minor improvements to enhance support for SG 2.0, but everything appears to be 👍 .

**Comment by dkichler** [comment]
Addressed by #262
